mm2src/mm2_main/tests/docker_tests/solana_tests.rs

```diff
@@ -111,3 +112,48 @@ fn test_sign_verify_message_spl() {
 
     assert!(response.is_valid);
 }
+
+#[test]
+fn test_disable_solana_platform_coin_with_token() {
+    let mm = solana_supplied_node();
+    block_on(enable_solana_with_tokens(
+        &mm,
+        "SOL-DEVNET",
+        &["USDC-SOL-DEVNET"],
+        "https://api.devnet.solana.com",
+        false,
+    ));
+
+    let enable_spl = block_on(enable_spl(&mm, "ADEX-SOL-DEVNET"));
+    let enable_spl: RpcV2Response<EnableSplResponse> = json::from_value(enable_spl).unwrap();
+    assert_eq!(1, enable_spl.result.balances.len());
+
+    // Try to disable platform coin, SOL-DEVNET
+    let disable = block_on(mm.rpc(&json! ({
+        "userpass": mm.userpass,
+        "method": "disable_coin",
+        "coin": "SOL-DEVNET",
+    })))
+    .unwrap();
+    assert_eq!(disable.0, StatusCode::OK);
+
+    // Confirm that token, USDC-SOL-DEVNET is also disabled
+    let response = block_on(mm.rpc(&json! ({
+        "userpass": mm.userpass,
+        "method": "my_balance",
+        "coin": "USDC-SOL-DEVNET"
+    })))
+    .unwrap();
+    assert_eq!(response.0, StatusCode::INTERNAL_SERVER_ERROR);
+    assert!(response.1.contains("No such coin: USDC-SOL-DEVNET"));
+
+    // Confirm platform coin token ADEX-SOL-DEVNET is also disabled
```

Comment:
`// Confirm that token, ADEX-SOL-DEVNET is also disabled` ADEX-SOL-DEVNET is a token not a platform coin. You need to add a check that platform coin `SOL-DEVNET` is disabled too.

Reply:
noted